### PR TITLE
NNS1-3135: Add comparators for maturity and neuron state

### PR DESCRIPTION
**Pull request by dskloetd**
# Motivation

When implementing custom neurons table sorting, we want to support sorting by maturity and state as well.

# Changes

1. Add `compareByMaturity` which compares by the sum of available and staked maturity.
2. Add `compareByState` which defines a specific order of the states.

# Tests

1. Unit tests added.
2. Also added coverage for the existing compare function. They were previously tested with `sortNeurons` but that was removed in https://github.com/dfinity/nns-dapp/pull/5104

# Todos

- [ ] Add entry to changelog (if necessary).
not necessary

### CODE REVIEW

File: frontend/src/lib/utils/neurons-table.utils.ts [hunk 1]
```diff
@@ -119,10 +120,24 @@ export const compareByStake = createDescendingComparator(
...
+);
+
 export const compareByDissolveDelay = createDescendingComparator(
   (neuron: TableNeuron) => neuron.dissolveDelaySeconds
 );
 
+export const compareByState = createDescendingComparator(
+  (neuron: TableNeuron) =>
+    [
+      NeuronState.Spawning,
```
Comment (mstrasinskis) [inline frontend/src/lib/utils/neurons-table.utils.ts:134]: Not sure if it's important, but should the `NeuronState.Unspecified` be handled here?

**Review by mstrasinskis — APPROVED** [review]
Looks good, just a couple of minor notes.

File: frontend/src/lib/utils/neurons-table.utils.ts (hunk 1 above)
Comment (dskloetd) [reply frontend/src/lib/utils/neurons-table.utils.ts:134]: As you pointed out, this is not a valid state and just exists for protobuf compatility:
https://github.com/dfinity/ic-js/blob/63ae83b041ceb445e928cdacdac246f04cc82a1a/packages/nns-proto/proto/governance.proto#L139

But even if we get a neuron with a state that's not listed, `indexOf` will return `-1` and the neuron will be sorted accordingly without any errors.

File: frontend/src/tests/lib/utils/neurons-table.utils.spec.ts [hunk 2]
```diff
@@ -383,4 +388,142 @@ describe("neurons-table.utils", () => {
...
+      const neuron2 = {
+        ...mockTableNeuron,
+        stakedMaturity: 200_000_000n,
+      };
+
+      expect(compareByMaturity(neuron1, neuron2)).toBe(1);
+      expect(compareByMaturity(neuron2, neuron1)).toBe(-1);
+    });
+
+    it("should tream available and staked maturity as equivalent", () => {
```
[the remark this answers is not in the document]
Comment (dskloetd) [reply frontend/src/tests/lib/utils/neurons-table.utils.spec.ts]: Yes, thanks.

**Pull request merged**